tools/src/test/java/org/apache/kafka/tools/reassign/ReassignPartitionsCommandTest.java

```diff
@@ -0,0 +1,811 @@
+
+        // Check that the assignment has not yet been started yet.
+        Map<TopicPartition, PartitionReassignmentState> initialAssignment = new HashMap<>();
+
+        initialAssignment.put(foo0, new PartitionReassignmentState(asList(0, 1, 2), asList(0, 1, 3), true));
+        initialAssignment.put(bar0, new PartitionReassignmentState(asList(3, 2, 1), asList(3, 2, 0), true));
+
+        try (Admin admin = Admin.create(Collections.singletonMap(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, clusterInstance.bootstrapServers()))) {
+            waitForVerifyAssignment(admin, assignment, false,
+                    new VerifyAssignmentResult(initialAssignment));
+
+            // Execute the assignment
+            runExecuteAssignment(false, assignment, -1L, -1L);
+            assertEquals(unthrottledBrokerConfigs, describeBrokerLevelThrottles(admin, unthrottledBrokerConfigs.keySet()));
+            Map<TopicPartition, PartitionReassignmentState> finalAssignment = new HashMap<>();
+            finalAssignment.put(foo0, new PartitionReassignmentState(asList(0, 1, 3), asList(0, 1, 3), true));
+            finalAssignment.put(bar0, new PartitionReassignmentState(asList(3, 2, 0), asList(3, 2, 0), true));
+
+            VerifyAssignmentResult verifyAssignmentResult = runVerifyAssignment(admin, assignment, false);
+            assertFalse(verifyAssignmentResult.movesOngoing);
+
+            // Wait for the assignment to complete
+            waitForVerifyAssignment(admin, assignment, false,
+                    new VerifyAssignmentResult(finalAssignment));
+
+            assertEquals(unthrottledBrokerConfigs,
+                    describeBrokerLevelThrottles(admin, unthrottledBrokerConfigs.keySet()));
+        }
+
+        // Verify that partitions are removed from brokers no longer assigned
+        verifyReplicaDeleted(new TopicPartitionReplica(foo0.topic(), foo0.partition(), 2));
+        verifyReplicaDeleted(new TopicPartitionReplica(bar0.topic(), bar0.partition(), 1));
+    }
+
+    private final Map<Integer, Map<String, Long>> unthrottledBrokerConfigs = new HashMap<>(); {
+        IntStream.range(0, 4).forEach(brokerId -> {
+            Map<String, Long> brokerConfig = new HashMap<>();
+
+            BROKER_LEVEL_THROTTLES.forEach(throttle -> brokerConfig.put(throttle, -1L));
+
+            unthrottledBrokerConfigs.put(brokerId, brokerConfig);
+        });
+    }
+
+    private void verifyReplicaDeleted(TopicPartitionReplica topicPartitionReplica) throws InterruptedException {
+        try (Admin admin = Admin.create(Collections.singletonMap(CommonClientConfigs.BOOTSTRAP_SERVERS_CONFIG, clusterInstance.bootstrapServers()))) {
+            TestUtils.waitForCondition(
+                    () -> {
+                        try {
+                            TopicDescription topicDescription = admin.describeTopics(singleton(topicPartitionReplica.topic())).topicNameValues().get(topicPartitionReplica.topic()).get();
```

Comment:
We can use `assertDoesNotThrow` to convert the checked exception to unchecked version.